sunbeam-python/sunbeam/core/manifest.py

```diff
@@ -167,48 +134,147 @@ def merge(self, other: "SoftwareConfig") -> "SoftwareConfig":
         terraform: dict[str, TerraformManifest] = utils.merge_dict(
             copy.deepcopy(self.terraform), copy.deepcopy(other.terraform)
         )
-        extra = utils.merge_dict(copy.deepcopy(self.extra), copy.deepcopy(other.extra))
-        return SoftwareConfig(juju=juju, charms=charms, terraform=terraform, **extra)
+        return SoftwareConfig(juju=juju, charms=charms, terraform=terraform)
 
-    @property
-    def extra(self) -> dict:
-        """Allow storing extra data."""
-        if self.__pydantic_extra__ is None:
-            self.__pydantic_extra__ = {}
-        return self.__pydantic_extra__
 
+class FeatureConfig(pydantic.BaseModel):
+    pass
 
-class Manifest(pydantic.BaseModel):
-    deployment: dict = {}
+
+def _default_software_config() -> SoftwareConfig:
+    snap = Snap()
+    return SoftwareConfig(
+        charms={
+            charm: CharmManifest(channel=channel)
+            for charm, channel in MANIFEST_CHARM_VERSIONS.items()
+        },
+        terraform={
+            tfplan: TerraformManifest(source=snap.paths.snap / "etc" / tfplan_dir)
+            for tfplan, tfplan_dir in TERRAFORM_DIR_NAMES.items()
+        },
+    )
+
+
+def _str_serialize(value: Any | None) -> str | None:
+    if value is not None:
+        return str(value)
+    return None
+
+
+class CoreConfig(pydantic.BaseModel):
+    class _ProxyConfig(pydantic.BaseModel):
+        proxy_required: bool | None = None
+        http_proxy: str | None = None
+        https_proxy: str | None = None
+        no_proxy: str | None = None
+
+    class _BootstrapConfig(pydantic.BaseModel):
+        management_cidr: str | None = pydantic.Field(
+            default=None, description="Management network CIDR"
+        )
+
+    class _Addons(pydantic.BaseModel):
+        metallb: str | None = None
+
+    class _K8sAddons(pydantic.BaseModel):
+        loadbalancers: str | None = None
+
+    class _User(pydantic.BaseModel):
+        run_demo_setup: bool | None = None
+        username: str | None = None
+        password: str | None = None
+        cidr: str | None = None
+        nameservers: str | None = None
+        security_group_rules: bool | None = None
+        remote_access_location: typing.Literal["local", "remote"] | None = None
+
+    class _ExternalNetwork(pydantic.BaseModel):
+        nic: str | None = None
+        cidr: str | None = None
+        gateway: str | None = None
+        start: str | None = None
+        end: str | None = None
+        network_type: typing.Literal["vlan", "flat"] | None = None
+        segmentation_id: int | None = None
+
+    class _HostMicroCephConfig(pydantic.BaseModel):
+        osd_devices: str | None = None
```

Comment:
`osd_devices` can be list as well.

Reply:
This is changing the behavior of manifest, and I would like to do these kind of changes in a more targeted commit.